packages/cli-lib/lib/walk.js

```diff
@@ -7,7 +7,7 @@ const STAT_TYPES = {
   DIRECTORY: 'dir',
 };
 
-const generateFilePromise = (dir, file) => {
+const generateFilePromise = (dir, file, { shallow = false } = {}) => {
```

Comment:
I think I mentioned this elsewhere but I am not super fond of `walk` having a shallow mode since it is designed to be for walking a tree and Node.js has `readdir` built in. Perhaps we could have a separate utility for reading a single directory that shares lower-level functionality with `walk`?

Reply:
I took a shot at this by creating a `read.js` file that handles reading the files shallowly in a directory. "Walk" shares some of the logic in it. Is that what you had in mind? There are quite a few different ways that this could be split so I'm open to other suggestions!

Reply:
Looks legit. We are probably long overdue to add some unit tests covering `walk` (sorry). It has gotten a lot more complex that when it was first written.